---
title: "Efficient exploration of DepMap cancer data using the depmap R package"
author:
- name: Theo Killian
  affiliation: Center for Cancer Biology, VIB-KU Leuven, Leuven, Belgium.
  email: theo.killian@vib.be
- name: Laurent Gatto
  affiliation: Computational Biology and Bioinformatics Unit, de Duve Institute, UCLouvain, Brussels, Belgium.
  email: laurent.gatto@uclouvain.be
date: "`r Sys.Date()`"
output: BiocWorkflowTools::f1000_article
bibliography: refs.bib
---
<!-- https://docs.google.com/document/d/1u4nqIVCIf60rB12Z7Dg8y-12RbQCJCDX_FuGgoWOpN4/edit#heading=h.qjfqotwxge5n -->

# Abstract

The `depmap` package facilitates access in the R environment to data from the
DepMap project, a multi-year collaborative effort by the Broad Institute and
Wellcome Sanger Institute, mapping genetic and chemical dependencies and other
molecular biological measurements of over 1800 cancer cell lines. The `depmap`
package formats this data for use of popular R data analysis and visualizing
tools found in the `tidyverse`, such as `dplyr` and `ggplot2`. The `depmap`
package utilizes `ExperimentHub`, distributing the DepMap data from the Cloud.
Specific versions of datasets may be selectively downloaded, providing a
reproducible research framework to support utilizing this data. This paper
describes a workflow demonstrating how to access, explore and visualize the
DepMap data in R using this Bioconductor package.

# Keywords

cancer, cancer dependency, DepMap, ExperimentHub, data mining,
reproducible research, Bioconductor

# Introduction

Genomic or molecular perturbations upon gene networks of cancerous cells may
result in metabolic vulnerabilities, (hereafter termed "dependencies") not
present in phenotypically healthy cells. An example of genetic dependency is
illustrated by a gene whose presence is not necessary for the maintenance of
normal metabolism and survival healthy cells, whereas the same gene is
determined to be "essential" for the vitality for a particular cancer cell
line. To clarify, knockout, knockdown or drug targeting of essential gene
dependencies are typically deleterious or lethal for many cancer cell types,
but not necessarily so for healthy cells. The prospect of selectively killing
cancer cells while potentially leaving healthy tissue unharmed, has long been
a goal of cancer therapeutic medicine. Accordingly, cancer dependencies are of
great interest to this field of research as potential drug targets. However,
the exact biological relationships of the majority of genetic dependencies of
cancer are not completely understood [@tsherniak2017defining]. One possible
theoretical explanation for the phenomena of gene dependency suggests that
essential genes are more likely to be central "hubs" within interaction
networks, whereas in healthy cells, the same gene may not be as centrally
located within the same network, as other parallel metabolic pathways are
available [@jeong2001lethality; @ma2003reconstruction; @viacava2021centrality].
In order to further elucidate the biological relationships of genetic
dependencies, it is requisite to build a "map" that illustrates the
association between these dependencies and the multiomic features of cancer.
The Cancer Dependency Map, better known as DepMap, represents such a map. The
Depmap Project is a collaborative initiative between the Broad Institute and
the Wellcome Sanger Institute, which aims to map genetic dependencies in a
broad range of cancer cell lines, as well as advancing understanding of the
molecular biology of cancer and to develop new drug targets and therapies in
precision cancer medicine [@broad2020depmap].

As of the most current DepMap release, over 1800 human cancer cell lines have
been mapped for genetic dependencies, a number that increases with each release.
New versions of select "core" datasets, such as CRISPR dependency screens and
inferred copy number are published on a regular release schedule. Prior to
`22Q2`, datasets were were released on a quarterly basis, but as `22Q2`, this
schedule will be reduced to a bi-annual release [@broad2020depmap].

The DepMap project utilizes CRISPR gene knockout screens as the primary method
to validate genetic dependencies in cancer cell lines [@broad2020depmap;
@meyers2017computational; @dempster2019extracting; @dempster2019agreement].
Genetic dependency scores displayed in the DepMap data are derived from the
observed log fold change in the amount of sgRNA detected in pooled cancer cell
lines after individual gene knockout [@cowley2014parallel;
@mcfarland2018improved]. To correct for potential off-target effects of gene
knockout which may overestimate dependency, the DepMap project utilizes the
CHRONOS (formerly CERES) algorithm to moderate the final estimated genomic
dependency estimate [@meyers2017computational; @dempster2021chronos]. In early
DepMap releases, RNAi gene knockdown was also employed to interrogate genetic
dependencies, but due to advancements in the accuracy of algorithms to account
for CRISPR off-target effects (i.e. CERES after the `19Q3` release and CHRONOS
after the `21Q1` release), the RNAi dependency measurements have been rendered
obsolete, and further data releases of RNAi datasets to describe gene
dependency have been discontinued since the `19Q3` release [@broad2020depmap;
@dempster2019extracting]. In addition to genomic dependency measurements of
cancer cell lines, chemical dependencies were also measured through the DepMap
PRISM viability screens that tested 4,518 compounds against 578 cancer cell
lines [@corsello2019non; @broad2020depmap]. A new proteomic dataset was added
with the `20Q2` release, that provides normalized quantitative profiling of
proteins of 375 cancer cell lines by mass spectrometry
[@nusinow2020quantitative]. The DepMap project has also compiled additional
datasets detailing molecular biological characterization of cancer cell lines,
such as genomic copy number (inferred from WGS, WES, and SNP array), Reverse
Phase Protein Array (RPPA) data, TPM gene expression data for protein coding
genes and genomic mutation data describing somatic point mutations and indels
called in the DepMap cell lines. All datasets all are publicly available under
[CC BY 4.0](https://creativecommons.org/licenses/by/4.0/) licence
[@broad2020depmap].

A table of the most recent datasets available for the `depmap` package
(as of `22Q2` release) is displayed in table \@ref(tab:datatab).

```{r datatab, message=FALSE, warning=FALSE, echo=FALSE, results='asis'}
library("kableExtra")
read.csv("./depmap_datasets_list.csv") %>%
    dplyr::select(-c(SourceUrl, BiocVersion)) %>%
    dplyr::filter(Dataset %in% c("crispr_22Q2", "TPM_22Q2", "copyNumber_22Q2",
                                 "mutationCalls_22Q2", "metadata_22Q2",
                                 "drug_sensitivity_21Q2", "rnai_19Q3",
                                 "RPPA_19Q3", "proteomic_20Q2",
                                 "achilles_22Q2", "gene_summary_22Q1")) %>%
    dplyr::mutate(Dataset = sub("_[0-9][0-9]Q[0-9]$", "", Dataset)) %>%
    dplyr::rename(Release = SourceVersion) %>%
    dplyr::arrange(desc(Release)) %>%
    knitr::kable(format = "latex",
                 booktabs = TRUE,
                 caption = paste0("Datasets available the depmap package. ",
                                  "The 'Release' column indicates the most ",
                                  "recent available release version.")) %>%
        kable_styling(full_width = TRUE,
                      latex_options = "striped",
                      font_size = 7) %>%
        column_spec(2, width = "5cm")
```

The `depmap` Bioconductor package was created with the intention of assisting
researchers in maximally extracting insights these rich datasets, facilitated
by importing the data into the R environment. The choice of porting DepMap data
in this way affords the use of data manipulation and visualization packages in
the Tidyverse, such as `dplyr` [@wickham2020package] and `ggplot2`
[@wickham2011ggplot2]. The value added by the `depmap` Bioconductor package
includes adding consistent and clearly descriptive feature names to all
datasets, appending select datasets with relevant metadata, converting all
datasets to long format tibbles [@muller2017tibble], as well as adding the
unique CCLE `depmap_id` that corresponds to individual cancer cell lines. The
addition of the `depmap_id` supports the comparison of various omics features
across datasets.

As new versions of DepMap datasets are published on regular release schedule,
resulting in dozens of versions of available datasets, it is not feasible to
include all dataset files directly within the directory of the `depmap` R
package. To keep the package lightweight, the `depmap` package utilizes and
fully depends on the `ExperimentHub` package to store and retrieve all versions
of the DepMap data in the Cloud using Microsoft Azure [@ExperimentHub]. It is
not necessary to install or be familiar with Microsoft Azure in order to use
the `depmap` package. Instead, the `depmap` package contains accessor functions
to directly download and cache the most current datasets from the Cloud into
the local R environment. Specific versions of datasets can also be downloaded
independently, if desired. The `depmap` package was designed with the goal of
supporting reproducible research by ensuring datasets from all releases will
remain accessible in this manner. The `depmap` R package continues to be
actively developed and maintained and is available as part of Bioconductor:
[https://bioconductor.org/packages/depmap](https://bioconductor.org/packages/depmap).

# Use cases

Dependency scores are the omic features of primary interest in the DepMap
Project datasets. These measurements can be found in `crispr` datasets, as well
as the discontinued `rnai` datasets. The genetic dependency score can be
interpreted as an expression of how important a particular gene is for
maintaining vitality for a given cancer cell line. For example, a highly
negative dependency score is reflects a large negative log fold change in the
population of cancer cells after gene knockout (i.e. CRISPR) or knockdown (i.e.
RNAi), implying that a given cell line is highly dependent on that knocked out
gene. In contrast, genes with a positive dependency score suggests that knockout
of such a gene increases the number of cancer cells. Prominent examples of genes
with positive dependency scores are RAS and PTEN, which play roles in cell cycle
regulation and tumor suppression, respectively. Genes that possess highly
negative dependency scores are of interest because they may be promising targets
for research in cancer therapy. However, there are many genes with highly
negative dependency scores that are not selective to cancer cell lines, but are
pan-lethal to healthy "metabolically normal" cells as well. In order to remove
these pan-lethal genes, it may be necessary to cross-reference genetic
dependency scores with the list of common "essential" genes listed in the
`gene_summary` dataset. To additionally complement genetic dependency
measurements, the `drug_sensitivity` dataset contains chemical dependency scores
which can be interpreted in a manner broadly analogous to genetic dependency. In
this workflow, we will describe exploring and visualizing several DepMap
datasets, including those that contain information on genetic dependency.

Below, we start by loading the packages needed to run this workflow:

```{r load_libraries, message=FALSE, warning=FALSE, echo=TRUE}
library("AnnotationHub")
library("ExperimentHub")
library("depmap")
library("dplyr")
library("ggplot2")
library("stringr")
```

Collectively, the `depmap` datasets are too large to be included into a typical
R package, therefore these data are instead stored in the Cloud. There are two
ways to access the `depmap` datasets. The first such way calls on dedicated
accessor functions that download, cache and load the latest available dataset
into the R workspace. All datasets available to download through the `depmap`
package are shown below:

```{r EH1, eval=FALSE}
## download combined Achilles and Sanger CRISPR gene effect knockout 
## dependency screens
crispr <- depmap::depmap_crispr()

## download gene-level inferred copy number inferred from WGS, WES and SNP array
copyNumber <- depmap::depmap_copyNumber()

## download gene expression TPM values of protein coding genes for DepMap cell
## lines
TPM <- depmap::depmap_TPM()

## download Reverse Phase Protein Lysate Microarray protein expression levels
RPPA <- depmap::depmap_RPPA()

## download RNAi gene knockdown dependency screens (obsolete)
rnai <- depmap::depmap_rnai()

## download metadata for all of DepMap cancer models and cell lines
metadata <- depmap::depmap_metadata()

## download MAF file containing all somatic point mutations and indels called
## on DepMap cell lines 
mutationCalls <- depmap::depmap_mutationCalls()

## download PRISM chemical dependency screens
drug_sensitivity <- depmap::depmap_drug_sensitivity()

## download quantitative profiling of proteins by mass spectrometry for DepMap
## cancer cell linesa
proteomic <- depmap::depmap_proteomic()

## download metadata for DepMap cancer models and cell lines for Achilles
## CRISPR screens
achilles <- depmap::depmap_achilles()

## download common essential genes determined by CRISPR screens
gene_summary <- depmap::depmap_gene_summary()
```

The alternative route to accessing `depmap` datasets requires interacting
directly with `ExperimentHub` package. This manner of data access is necessary
for obtaining a specific version of a `depmap` dataset from any available
release. This alternative data access method uses the `ExperimentHub()`
function to creates an `ExperimentHub` object, which can be queried for
specific terms relevant to each dataset. The list of datasets available that
correspond to the query, "`depmap`" are shown below:

```{r EH2, message=FALSE, warning=FALSE, echo=TRUE}
## create ExperimentHub query object
eh <- ExperimentHub::ExperimentHub()
AnnotationHub::query(eh, "depmap")
```

By selecting a given dataset by their unique `EH` number within the
`ExperimentHub` query object, a specific datasets can be downloaded, cached
and loaded into the workspace as a tibble. By importing the `depmap` data into
the R environment, the data can be mined more effectively through the use of R
data manipulation tools, which will be explored and illustrated in the
following sections. Shown below, commonly used datasets from the `22Q2` release
are downloaded by their unique `EH` number.

```{r load_data, message=FALSE, warning=FALSE, echo=TRUE}
## download and cache required datasets into the R environment:
crispr           <- eh[["EH7523"]] ## cripsr_22Q2
copyNumber       <- eh[["EH7524"]] ## copyNumber_22Q2
TPM              <- eh[["EH7525"]] ## TPM_22Q2
mutationCalls    <- eh[["EH7526"]] ## mutationCalls _22Q2
metadata         <- eh[["EH7527"]] ## metadata_22Q2
gene_summary     <- eh[["EH7529"]] ## gene_summary_22Q1
# proteomic        <- eh[["EH3459"]] ## proteomic_20Q2
# drug_sensitivity <- eh[["EH7530"]] ## drug_sensitivity_21Q2
```

The `depmap` package makes it is possible to investigate the omic features and
identify potential molecular targets of specific cancer subtypes using functions
from the `depmap` and Tidyverse packages. For example: we will illustrate how
to investigate dependent genes in skin cancer for potential molecular targets
specific to this cancer subtype, while simultaneously filtering out pan-lethal
genes. Shown below, the `crispr` dataset is selected for cell lines with
*"SKIN"* in the CCLE name (corresponding to melanoma samples) and displaying a
list of genes with the highest dependency scores that are also found in the list
of common essential genes from the `gene_summary` dataset.

```{r, soft_tissue_cell_lines, echo=TRUE}
## display list of top 25 genes of CRISPR-derived dependency scores,
## cross-referenced with the gene summary data
crispr %>%
  dplyr::filter(stringr::str_detect(cell_line, "SKIN")) %>%
  dplyr::arrange(dependency) %>%
  dplyr::left_join(gene_summary, by = "gene_name") %>% ## add gene summary data
  dplyr::filter(common_essential == TRUE, ## remove pan-lethal hits
                dataset == "Chronos_Combined") %>%
  dplyr::select(cell_line, gene_name, dependency) %>%
  dplyr::slice(1:25)
```

Melanoma accounts for a significant fraction of global cancer diagnoses and is
rising in prevalence in developed, predominantly fair-skinned countries, growing
over 320% in the US since 1975. NRAS dependent melanoma is a subtype of melanoma
with a particularly poor prognosis. Mutations in NRAS occur in approximately a
fifth of cutaneous melanomas and are associated with aggressive clinical
behavior [@saginala2021epidemiology; munoz2017nras]. Shown below, the `crispr`
and `metadata` datasets are combined to display the dependency scores for NRAS
in primary versus metastatic melanoma. The average dependency scores for NRAS
across all cell lines is shown as a blue dotted line, whereas are the average
for primary melanoma is shown in red, and metastatic melanoma shown in gray. We
can observe that the mean dependency score for NRAS is slightly higher for
metastatic melanoma, but has a similar distribution as primary melanoma.
Additionally, melanoma is more dependent on NRAS than other types of cancer
(blue dotted line).

```{r histNRAS, message=FALSE, warning=FALSE, fig.cap="Histogram of CRISPR dependency scores for gene NRAS."}
## filter metadata for primary or metastasis melanoma samples
metadata %>%
  dplyr::filter(stringr::str_detect(cell_line, "SKIN"),
                primary_or_metastasis == "Primary") %>%
  dplyr::select(cell_line, primary_or_metastasis) -> meta_primary

metadata %>%
  dplyr::filter(stringr::str_detect(cell_line, "SKIN"),
                primary_or_metastasis == "Metastasis") -> meta_metastasis

## calculate mean NRAS dep scores across features
crispr %>% dplyr::filter(gene_name == "NRAS") -> mean_crispr_dep_all
crispr %>%
  dplyr::filter(cell_line %in% c(meta_primary$cell_line),
                gene_name == "NRAS") -> mean_crispr_dep_prim
crispr %>%
  dplyr::filter(cell_line %in% c(meta_metastasis$cell_line),
                gene_name == "NRAS") -> mean_crispr_dep_metast

## join metadata filtered on melanoma with crispr scores and plot histogram 
metadata %>%
  # dplyr::filter(stringr::str_detect(cell_line, "SKIN")) %>% 
  dplyr::filter(stringr::str_detect(lineage_subtype, "melanoma")) %>% 
  dplyr::select(cell_line, lineage_subtype, primary_or_metastasis) %>% 
  dplyr::left_join(crispr, by = "cell_line") %>% 
  dplyr::filter(gene_name == "NRAS",
                !is.na(primary_or_metastasis)) %>%
  ggplot(aes(x = dependency, fill = primary_or_metastasis)) +
  geom_histogram(position = "identity", binwidth = 0.05, alpha = 0.5) +
  geom_density(alpha = 0.5) +
  scale_fill_manual(values = c("#999999", "firebrick2")) +
  theme_minimal() +
  geom_vline(xintercept = mean(mean_crispr_dep_all$dependency, na.rm = TRUE),
             linetype = "dashed", color = "blue") +
  geom_vline(xintercept = mean(mean_crispr_dep_prim$dependency, na.rm = TRUE),
             linetype = "dashed", color = "firebrick2") +
  geom_vline(xintercept = mean(mean_crispr_dep_metast$dependency, na.rm = TRUE),
             linetype = "dashed", color = "#999999")
```

Activating mutations in RAS oncogenes (such as NRAS) are found in roughly a
third of all human cancers [@saginala2021epidemiology]. To put the NRAS
dependency data from melanoma in perspective, a more complex plot displaying
information from the `crispr` and `mutationCalls` datasets are shown below
\@ref(fig:NRAS). This illustrates the distribution of dependency scores for gene
`NRAS` for each major type of cancer, while highlighting the qualitative nature
of mutations of this gene in such cancer cell lines (e.g. if such mutations are
damaging or conserved, etc.). We observe in this visualization a high fraction 
of "other non-conserving mutations" (e.g. missense mutations) represented in the
highest dependency scores across all cancer types (light blue dots). This
suggests that certain types of tumors with specific NRAS mutations are very
dependent on this gene.

```{r NRAS, message=FALSE, warning=FALSE, fig.cap="Plot of CRISPR dependency scores for gene NRAS by lineage."}
## merge metadata with crispr dependencies and mutation calls for NRAS
metadata %>%
  dplyr::select(depmap_id, lineage) %>%
  dplyr::full_join(crispr, by = "depmap_id") %>%
  dplyr::filter(gene_name == "NRAS") %>%
  dplyr::full_join((mutationCalls %>%
                      dplyr::select(depmap_id, entrez_id, is_cosmic_hotspot,
                                    var_annotation)),
                   by = c("depmap_id", "entrez_id")) -> m_c_mu

## plot merged data for NRAS
m_c_mu %>%
    ggplot(aes(x = dependency, y = lineage)) +
    geom_point(alpha = 0.4, size = 0.5) +
    geom_point(data = subset(m_c_mu, var_annotation == "damaging"),
               color = "hotpink") +
    geom_point(data = subset(m_c_mu, var_annotation == "other non-conserving"),
               color = "dodgerblue") +
    geom_vline(xintercept = mean(m_c_mu$dependency, na.rm = TRUE),
               linetype = "dotted", color = "red") +
    theme_minimal()
```

However, gene mutations are merely one aspect to consider to further
characterize gene dependencies. Many cancer phenotypes are the result of changes
in gene expression [@li2019changes; @hernandez2019many; @felts2019stochastic].
The extensive coverage of the `depmap` data affords visualization of genetic
expression patterns across many major types of cancer. Figure \@ref(fig:tpm)
below shows a boxplot illustrating expression values for gene `NRAS` by cancer
lineage for both primary and metastatic tumors:

```{r tpm, message=FALSE, warning=FALSE, fig.cap="Boxplot of TPM expression values for gene NRAS by lineage."}
metadata %>%
  dplyr::select(depmap_id, lineage, primary_or_metastasis) %>%
  dplyr::full_join(TPM, by = "depmap_id") %>%
  dplyr::filter(gene_name == "NRAS",
                primary_or_metastasis %in% c("Primary", "Metastasis")) %>%
  ggplot(aes(x = lineage, y = rna_expression, fill = lineage)) +
  geom_boxplot(outlier.alpha = 0.1) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none") + 
  theme_minimal() +
  facet_grid(rows = vars(primary_or_metastasis))
```

Elevated gene expression and genetic dependency in cancer cell lines have been
reported in literature [@tsherniak2017defining; @mcfarland2018improved].
Therefore, genes with elevated gene expression and high genetic dependency may
present especially interesting research targets. Figure \@ref(fig:melanoma2)
shows a plot of expression versus CRISPR gene dependency for all melanoma
measurements. The vertical line represents the average gene expression for
this form of cancer, while the horizontal line represents the average dependency
for this cancer type. 

```{r melanoma1, message=FALSE, warning=FALSE}
metadata %>%
  dplyr::select(depmap_id, cell_line, primary_disease, subtype_disease) %>%
  dplyr::filter(primary_disease == "Skin Cancer",
                subtype_disease == "Melanoma") -> melanoma

crispr %>%
  dplyr::select(depmap_id, gene, gene_name, dependency) -> crispr_sub

TPM %>%
  dplyr::select(depmap_id, gene, gene_name, rna_expression) -> tpm_sub

melanoma %>%
  dplyr::left_join(crispr_sub, by = "depmap_id") %>%
  dplyr::select(-c(cell_line, primary_disease, subtype_disease, gene_name)
                ) -> melanoma_dep

melanoma %>%
  dplyr::left_join(tpm_sub, by = "depmap_id") -> melanoma_exp

melanoma_dep %>%
  dplyr::full_join(melanoma_exp, by = c("depmap_id", "gene")) %>%
  dplyr::filter(!is.na(rna_expression)) -> melanoma_dat_exp
```

```{r melanoma2, message=FALSE, warning=FALSE, fig.cap = "Expression vs crispr gene dependency for Melanoma."}
melanoma_dat_exp %>%
    ggplot(aes(x = dependency, y = rna_expression)) +
    geom_point(alpha = 0.4, size = 0.5) +
    geom_vline(xintercept = mean(melanoma_dat_exp$dependency, na.rm = TRUE),
               linetype = "dotted", color = "red") +
    geom_hline(yintercept = mean(melanoma_dat_exp$rna_expression, na.rm = TRUE),
               linetype = "dotted", color = "red") +
    theme(axis.text.x = element_text(angle = 45)) +
    theme_minimal()
```

We can observer that almost all of the genes with the highest dependency scores
in the far left of the plot above \@ref(fig:melanoma2) display increased
expression above the average dependency for this cancer type. Below, we sort and
display genes with the highest dependency scores and highest TPM
gene expression which are found in the upper left section of the plot above.

```{r}
melanoma_dat_exp %>%
  dplyr::select(cell_line, gene_name, dependency, rna_expression) %>%
  dplyr::arrange(dependency, rna_expression) %>% 
  dplyr::slice(1:1000)
```

In addition to changes in gene expression, genomic copy number gain or loss may
also play a role in some cancer phenotypes [@meyers2017computational;
@aguirre2016genomic; @shao2019copy]. Figure \@ref(fig:violinCNVNRAS) shows the
primary and metastatic melanoma cell lines that have NRAS copy gain greater than
1:

```{r violinCNVNRAS, message=FALSE, warning=FALSE, fig.cap="Boxplot of log copy number for gene NRAS by lineage"}
metadata %>%
  dplyr::filter(primary_disease == "Skin Cancer",
                subtype_disease == "Melanoma",
                primary_or_metastasis %in% c("Primary", "Metastasis")) %>%
  dplyr::select(-cell_line) %>%
  dplyr::left_join(copyNumber, by = "depmap_id") %>%
  dplyr::filter(gene_name == "NRAS",
                !is.na(cell_line)) %>%
  ggplot(aes(x = primary_or_metastasis, y = log_copy_number,
             fill = primary_or_metastasis)) +
  geom_violin(alpha = 0.8) +
  geom_jitter(alpha = 0.8, width = 0.25, height = 0.25, size = 0.8) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme_minimal() +
  theme(legend.position = "none") + 
  ggrepel::geom_text_repel(#nudge_x = 0.1, nudge_y = 0.1,
    aes(label = ifelse(log_copy_number > 1, as.character(cell_line), '')))
```

The broad scope of the Depmap data allows the display of log genomic copy number
for across many cancer lineages. Figure \@ref(fig:boxCNVNRAS) shows such a plot
for gene `NRAS` for each major type of cancer lineage:

```{r boxCNVNRAS, message=FALSE, warning=FALSE, fig.cap="Boxplot of log copy number for gene NRAS by lineage"}
metadata %>%
  dplyr::select(depmap_id, lineage) %>%
  dplyr::full_join(copyNumber, by = "depmap_id") %>%
  dplyr::filter(gene_name == "NRAS",
                lineage != "NA",
                !grepl("engineered", lineage)) %>%
  ggplot(aes(x = lineage, y = log_copy_number, fill = lineage)) +
  geom_boxplot(outlier.alpha = 0.1) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylim(0, 2.5) +
  theme(legend.position = "none")
```

# Discussion and outlook

We hope that this package will be used by cancer researchers to efficiently
explore the DepMap data and to support their research. Additionally, we highly
encourage future `depmap` users to combine `depmap` data with other datasets of
interest, such as `TCGA`
[https://portal.gdc.cancer.gov](https://portal.gdc.cancer.gov) and other `CCLE`
datasets:
[https://sites.broadinstitute.org/ccle/datasets](https://sites.broadinstitute.org/ccle/datasets)

The depmap R package will continue to be maintained in line with Bioconductor
releases, along with the biannual updates of the DepMap datasets.

We welcome feedback and questions from the community. We also highly appreciate
contributions to the code in the form of pull requests.

# Software availability

All packages used in this workflow are available from the Comprehensive R
Archive Network [https://cran.r-project.org](https://cran.r-project.org) or
Bioconductor [http://bioconductor.org](http://bioconductor.org). The specific
version numbers of R and the packages used are shown below under the section
on `sessionInfo`.

Note: the latest `depmap` datasets are only available and visible in a
`ExperimentHub` query of the local environment is using the latest `devel`
versions of Bioconductor and R. To install the `devel` version of Bioconductor:

```{r message=FALSE, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "devel")
BiocManager::valid()              # checks for out of date packages
```

To install the `depmap` package:

```{r message=FALSE, eval=FALSE}
BiocManager::install('depmap')
```

```{r echo = FALSE}
sessionInfo()
```

# Acknowledgements

# Competing interests

No competing interests were disclosed.

# Grant information

# References
