---
title: "Exploiting the Depmap cancer dependency data using the depmap R package"
author:
- name: Theo Killian
  affiliation: Computational Biology, UCLouvain
- name: Laurent Gatto
  affiliation: Computational Biology, UCLouvain
date: "`r Sys.Date()`"
output: BiocWorkflowTools::f1000_article
bibliography: refs.bib
---


**R version**: `r R.version.string`

**Bioconductor version**: `r BiocManager::version()`

**Package**: `r packageVersion("depmap")` -->

# Abstract

The `depmap` package facilitates access in the R environment to the data from
the Depmap project, a multi-year collaborative effort by the Broad Institute and
Wellcome Sanger Institute, mapping genetic and chemical dependencies and other
molecular biological measurements of over 1700 cancer cell lines. The `depmap`
package formats this data for use of popular R data analysis and visualizing
tools such as `dplyr` and `ggplot2` to represent and visualize these rich
datasets. In addition, the `depmap` package utilizes `ExperimentHub`, storing
versions of the Depmap data accessible from the Cloud, which may be selectively
downloaded, providing a reproducible research framework to support exploiting
this data. This paper describes a workflow demonstrating how to access and
visualize the Depmap data in R using this package.

# Keywords

cancer, cancer dependency, Depmap, R, dplyr, ggplot2, ExperimentHub, Cloud, data
mining, reproducible research, Bioconductor

# Introduction

The consequences of genomic alterations of cancer cells on the molecular
biological landscape of the cell may result in differential vulnerabilities, or
"dependencies" compared to those of healthy cells. An example may be a gene not
necessary for the survival in healthy cells, but essential for the vitality of
particular cancer cell line. The exact nature of many of these dependencies in
cancer cell lines is not completely understood. A map illustrating the
relationships between the genetic features of cancer and those of cancer
dependencies is desirable. The Cancer Dependency Map or "Depmap", a
collaborative initiative between the Broad Institute and the Wellcome Sanger
Institute, aims to map such dependencies in a broad range cancer cell lines,
intended to mirror the distribution of various cancer diseases in the general
population, with the intention of exploiting this knowledge to develop new
therapies in precision cancer medicine.

The Depmap initative is, as of the date of this publication, an ongoing project,
with new data releases of select datasets every 90 days. As of the most current
`20Q1` Depmap release, 1775 human cancer cell lines have been mapped for
dependencies. The primary method utilized in the Depmap project to map genomic
dependencies is gene knockout performed by CRISPR. Genetic dependency is
calculated from the observed log fold change in the amount of shRNA detected
after gene knockout. To correct for potential off-target effects of gene
knockout in overestimating dependency with CRISPR, the Depmap iniative utilized
the CERES algorithm to moderate the final dependency estimation. It should be
noted that due to advancements in the CERES algorithm to account for seed
effects, the RNAI dependency has been rendered redundant, and further data
releases for this dependency measurement have been discontinued as of `19Q3`
[@stuff]. In addition genomic dependency measurements of cancer cell lines,
chemical dependencies were taken via Depmap PRISM viability screens that as of
the `20Q1` release, consisted of 4,518 compounds tested against 578 cancer cell
lines. The Depmap project has also compiled additional datasets detailing
molecular biological characterization of cancer cell lines, such as genomic copy
number, Reverse Phase Protein Array (RPPA) data, TPM gene expression data for
protein coding genes and genomic mutation data. These datasets are updated
quarterly on a release schedule and are publically available under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/) licence.

The `depmap` Bioconductor package was created in order to maximally exploit
these rich datasets and to aide reproducible research, by importing the data
into the R environment. The Depmap datasets were cleaned by converting all
datasets to the long format, as well as adding the unique key `depmap_id` for
all data tables, in order to make features more comparable, facilitating the use
of common R packages such as `dplyr` and `ggplot2`.

As new Depmap datasets are released on a quarterly basis, it is not feasible to
include all dataset files in binary directly within the directory of the
`depmap` R package. To keep the package lightweight, the `depmap` package
utilizes and fully depends on the `ExperimentHub` package to store and retrieve
all versions of the Depmap data (starting from `19Q1` through `20Q1`) in the
Cloud using AWS. The `depmap` package contains accessor functions to directly
download and cache the most current datasets from the Cloud into the R
environment. Specific datasets, such as older datasets, which have been used in
prior research can also be downloaded, if desired. This feature has the added
advantage of enhancing reproducible research, such that specific versions of
Depmap data can be selected, in addition to having access to the most current
datasets. The `depmap` R package is available as part of Bioconductor at:
https://bioconductor.org/packages/devel/data/experiment/html/depmap.html

# Use cases

The features of primary interest from the Depmap Project are the measurements of
cancer gene dependency scores, found in datasets `rnai` and `crispr`. The
dependency score is an expression of how vital a particular gene for a given
cancer cell line is in terms of the lethality resulting from the knockout or
knockdown of that gene. For example, a highly negative dependency score is
derived from a large negative log fold change in the population of cancer cells
after gene knockout or knockdown, implying that a given cell line is highly
dependent on that gene. Genes that possess such highly negative dependency
scores may be interesting targets for research in cancer medicine.

```{r load_libraries, message=FALSE, warning=FALSE, echo=TRUE}
## load libraries
library("dplyr")
library("ggplot2")
library("tibble")
library("tidyr")
library("gridExtra")
library("stringr")
library("depmap")
library("ExperimentHub")
```

The `depmap` datasets are too large to be included into the binary of a typical
R package, therefore this data is stored in the Cloud. There are two ways to
access the `depmap` datasets from the Cloud. The first such way calls on
dedicated accessor functions that download and cache the latest available
dataset into the R environment. An example is shown below:

```{r EH, message=FALSE, warning=FALSE, echo=TRUE, eval=FALSE}
rnai <- depmap_rnai()
crispr <- depmap_crispr()
copyNumber <- depmap_copyNumber()
TPM <- depmap_RPPA()
RPPA <- depmap_TPM()
metadata <- depmap_metadata()
mutationCalls <- depmap_mutationCalls()
drug_sensitivity <- depmap_drug_sensitivity()
```

Alternatively, a specific dataset (from any available release) can be accessed
with the `ExperimentHub()` accessor function, which creates an `ExperimentHub`
object, which can be queried for specific terms. The list of datasets available
that correspond to the query, `depmap` are shown below:

```{r EH, message=FALSE, warning=FALSE, echo=TRUE}
## create ExperimentHub query object
eh <- ExperimentHub()
query(eh, "depmap")
```

Specific datasets are downloaded via their unique `EH` numbers and cached into
the R environment. Shown below, datasets from the `19_Q3` release are downloaded
in this way via their such as the unique `EH` numbers that corresponds each
dataset.

```{r load_data, message=FALSE, warning=FALSE, echo=TRUE}
## download and cache select datasets
rnai <- eh[["EH3080"]]
mutationCalls <- eh[["EH3085"]]
metadata <- eh[["EH3086"]]
crispr <- eh[["EH3081"]]
TPM <- eh[["EH3084"]]
copyNumber <- eh[["EH3082"]]
drug_sensitivity <- eh[["EH3087"]]
```

By importing the `depmap` data into the R environment, the data can be mined
more effectively. For example, if one was interested in researching soft tissue
sarcomas and wanted to search all such cancer cell lines for the gene with the
greatest dependency, it is possible to accomplish this task by using data
manipulation and visualization tools `dplyr` and `ggplot2`. Below, the `crispr`
dataset is selected for cell lines with *"SOFT_TISSUE"* in the CCLE name, and
displaying a list of the highest dependency scores.

```{r, soft_tissue_cell_lines, echo=TRUE}
## list of dependency scores
crispr %>% dplyr::select(cell_line, gene_name, dependency) %>%
           dplyr::filter(stringr::str_detect(cell_line, "SOFT_TISSUE")) %>%
           dplyr::arrange(dependency) %>% 
           head(10)
```

As the gene `RPL14` appears several times in the top dependencies scores, it may
make an interesting candidate target. Below, a plot of the `crispr` data is
displayed as a histogram showing the distribution of dependency scores for gene
`RPL14`. The red dotted line signifies the mean dependency score for that gene.

```{r message=FALSE, warning=FALSE}
## Basic histogram
mean_crispr_dep <- crispr %>% dplyr::select(gene_name, dependency) %>% 
                              dplyr::filter(gene_name == "RPL14")

crispr %>% dplyr::select(gene, gene_name, dependency) %>% 
           dplyr::filter(gene_name == "RPL14") %>% 
           ggplot(aes(x = dependency)) + geom_histogram() +
           geom_vline(xintercept = mean(
           mean_crispr_dep$dependency, na.rm = TRUE), linetype = "dotted",
                      color = "red") +
           ggtitle("Histogram of CRISPR dependency scores for gene RPL14")
```

A more complex plot of the `crispr` data, as shown below involves plotting the
distribution of dependency scores for gene `RPL14` for each major type of
cancer, while highlighting the nature of mutations of this gene in such cancer
cell lines (e.g. if such are COSMIC hotspots, damaging, etc.). Notice that the
plot above reflects the same overall distribution in two dimensions.

```{r message=FALSE, warning=FALSE}
meta_crispr <- metadata %>%
             dplyr::select(depmap_id, lineage) %>%
             dplyr::full_join(crispr, by = "depmap_id") %>%
             dplyr::filter(gene_name == "RPL14") %>% 
             dplyr::full_join((mutationCalls %>%
             dplyr::select(depmap_id, entrez_id,
                    is_cosmic_hotspot, var_annotation)),
                    by = c("depmap_id", "entrez_id"))

meta_crispr %>%
    ggplot(aes(x = dependency, y = lineage)) +
    geom_point(alpha = 0.4, size = 0.5) +
    geom_point(data = subset(meta_crispr,
                             var_annotation == "damaging"), color = "red") +
    geom_point(data = subset(meta_crispr,
                             var_annotation == "other non-conserving"),
                             color = "blue") +
    geom_vline(xintercept = mean(meta_crispr$dependency, na.rm = TRUE),
               linetype = "dotted", color = "red") +
    ggtitle("Plot of CRISPR dep. scores for gene RPL14 by lineage")
```

Below is a boxplot displaying expression values for gene `RPL14` by lineage:

```{r message=FALSE, warning=FALSE}
metadata %>%
      dplyr::select(depmap_id, lineage) %>%
      dplyr::full_join(TPM, by = "depmap_id") %>%
      dplyr::filter(gene_name == "RPL14") %>% 
      ggplot(aes(x = lineage, y = expression, fill = lineage)) +
      geom_boxplot(outlier.alpha = 0.1) +
      ggtitle("Boxplot of expression values for gene RPL14 by lineage") +
      theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
      theme(legend.position = "none")
```

High dependency, high expression genes are more likely to interesting research
targets. Below is a plot of expression vs CRISPR gene dependency for
Rhabdomyosarcoma Sarcoma:

```{r  message=FALSE, warning=FALSE}
## expression vs crispr gene dependency for Rhabdomyosarcoma Sarcoma
sarcoma <- metadata %>%
    dplyr::select(depmap_id, cell_line, primary_disease, subtype_disease) %>%
    dplyr::filter(primary_disease == "Sarcoma",
                  subtype_disease == "Rhabdomyosarcoma")

crispr_sub <- crispr %>% dplyr::select(depmap_id, gene, gene_name, dependency)

tpm_sub <- TPM %>% dplyr::select(depmap_id, gene, gene_name, expression)

sarcoma_dep <- sarcoma %>% dplyr::left_join(crispr_sub, by = "depmap_id") %>%
                           dplyr::select(-cell_line, -primary_disease,
                                         -subtype_disease, -gene_name)

sarcoma_exp <- sarcoma %>% dplyr::left_join(tpm_sub, by = "depmap_id")

sarcoma_dat_exp <- dplyr::full_join(sarcoma_dep, sarcoma_exp,
                                    by = c("depmap_id", "gene")) %>%
                   dplyr::filter(!is.na(expression))

p3 <- ggplot(data = sarcoma_dat_exp, aes(x = dependency, y = expression)) +
      geom_point(alpha = 0.4, size = 0.5) +
      geom_vline(xintercept = mean(sarcoma_dat_exp$dependency, na.rm = TRUE),
                 linetype = "dotted", color = "red") +
      geom_hline(yintercept = mean(sarcoma_dat_exp$expression, na.rm = TRUE),
                 linetype = "dotted", color = "red") +
      ggtitle("Scatterplot of CRISPR dependency vs expression values for gene")
p3 + theme(axis.text.x = element_text(angle = 45))
```

A selection of the genes shown above with the lowest depenency scores, which can
be cross-referenced by TPM gene expression. Such genes may present an example as
interesting research targets.

```{r}
sarcoma_dat_exp %>%
    dplyr::select(cell_line, gene_name, dependency, expression) %>%
    dplyr::arrange(dependency) %>% 
    head(10)
```

Below is a boxplot displaying log genomic copy number for gene `RPL14` by
lineage:

```{r message=FALSE, warning=FALSE}
metadata %>%
      dplyr::select(depmap_id, lineage) %>%
      dplyr::full_join(copyNumber, by = "depmap_id") %>%
      dplyr::filter(gene_name == "RPL14") %>%
      ggplot(aes(x = lineage, y = log_copy_number, fill = lineage)) +
      geom_boxplot(outlier.alpha = 0.1) +
      ggtitle("Boxplot of log copy number for gene RPL14 by lineage") +
      theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
      theme(legend.position = "none")
```

# Discussion and Outlook

We hope that this package will be used by cancer researchers to dig deeper into
the Depmap data and to support their research.

The depmap R package will continue to be maintained in line with the biannual
Bioconductor release, in addition to quarterly releases of Depmap data.

We welcome feedback and questions from the community. We also highly appreciate
contributions to the code in the form of pull requests.

# Software availability

Please note that this manuscript uses version 1.12 of the depmap R package,
which is in the development version of Bioconductor (v.3.10). There are two
possible ways of installing it:

1) Install the development version of R (v.3.6) — required for Bioconductor
v.3.10 — and install depmap using the command:

```{r message=FALSE, warning=FALSE, eval=FALSE}
BiocManager::install('depmap', version = 'devel')
```

–or–
    
2) Install depmap 1.0 directly from the github repository using the devtools R
package:

```{r message=FALSE, warning=FALSE, eval=FALSE}
install.packages('devtools')
library(devtools)
install_github('uclouvain-cbio/depmap')
```

# Session information

```{r echo = FALSE}
sessionInfo()
```

<!-- # Acknowledgements -->

# Competing interests

No competing interests were disclosed.

# Grant information

# References

<!-- 1. [Tsherniak, Aviad, et al. "Defining a cancer dependency map." Cell 170.3 (2017): -->
<!-- 564-576.](https://www.ncbi.nlm.nih.gov/pubmed/28753430). -->

<!-- 2. Depmap, A Cancer Dependency Map to systematically identify genetic and -->
<!-- pharmacologic dependencies and the biomarkers that predict them. 2019. Retrieved -->
<!-- from https://depmap.org/portal/download/ -->

<!-- 3. McFarland, J. M., Ho, Z. V., Kugener, G., Dempster, J. M., Montgomery, P. G., -->
<!-- Bryan, J. G., ... & Golub, T. R. (2018). Improved estimation of cancer -->
<!-- dependencies from large-scale RNAi screens using model-based normalization and -->
<!-- data integration. Nature communications, 9. -->

<!-- 4. Meyers, R. M., Bryan, J. G., McFarland, J. M., Weir, B. A., Sizemore, A. E., -->
<!-- Xu, H., ... & Goodale, A. (2017). Computational correction of copy number effect -->
<!-- improves specificity of CRISPR–Cas9 essentiality screens in cancer cells. Nature -->
<!-- genetics, 49(12), 1779. -->

<!-- 5. depmap. Cancer Dependency Map Data Package. 2019. Retrieved -->
<!-- from https://bioconductor.org/packages/devel/data/experiment/html/depmap.html -->

<!-- 6. R language -->
