---
title: "Exploiting the Depmap cancer dependency data using the depmap R package"
author:
- name: Theo Killian
  affiliation: Computational Biology and Bioinformatics Unit, de Duve Institute, UCLouvain, Brussels, Belgium
- name: Laurent Gatto
  affiliation: Computational Biology and Bioinformatics Unit, de Duve Institute, UCLouvain, Brussels, Belgium
date: "`r Sys.Date()`"
output: BiocWorkflowTools::f1000_article
bibliography: refs.bib
---


# Abstract

The `depmap` package facilitates access in the R environment to the
data from the Depmap project, a multi-year collaborative effort by the
Broad Institute and Wellcome Sanger Institute, mapping genetic and
chemical dependencies and other molecular biological measurements of
over 1700 cancer cell lines. The `depmap` package formats this data
for use of popular R data analysis and visualizing tools such as
`dplyr` and `ggplot2`. In addition, the `depmap` package utilizes
`ExperimentHub`, storing versions of the Depmap data accessible from
the Cloud, which may be selectively downloaded, providing a
reproducible research framework to support exploiting this data. This
paper describes a workflow demonstrating how to access and visualize
the Depmap data in R using this package.

# Keywords

cancer, cancer dependency, Depmap, ExperimentHub, data mining, 
reproducible research, Bioconductor

# Introduction

The consequences of genomic alterations of cancer cells on the molecular
biological landscape of the cell may result in differential vulnerabilities, or
"dependencies" compared to those of healthy cells. An example may be a gene not
necessary for the survival in healthy cells, but essential for the vitality of
particular cancer cell line. The exact nature of many of these dependencies in
cancer cell lines is not completely understood [@tsherniak2017defining]. A map
illustrating the relationships between the genetic features of cancer and those
of cancer dependencies is desirable. The Cancer Dependency Map or "Depmap", a
collaborative initiative between the Broad Institute and the Wellcome Sanger
Institute, aims to map such dependencies in a broad range cancer cell lines,
intended to mirror the distribution of various cancer diseases in the general
population, with the intention of exploiting this knowledge to develop new
therapies in precision cancer medicine [@broad2020depmap].

The Depmap initative is, as of the date of this publication, an
ongoing project, with new data releases of select datasets every 90
days. As of the most current `20Q1` Depmap release, 1775 human cancer
cell lines have been mapped for dependencies [@broad2020depmap]. The
primary method utilized in the Depmap project to map genomic
dependencies is gene knockout performed by CRISPR [@broad2020depmap;
@meyers2017computational; @dempster2019extracting;
@dempster2019agreement].  Genetic dependency is calculated from the
observed log fold change in the amount of shRNA detected after gene
knockout [@cowley2014parallel; @mcfarland2018improved].  To correct
for potential off-target effects of gene knockout in overestimating
dependency with CRISPR, the Depmap iniative utilized the CERES
algorithm to moderate the final dependency estimation
[@meyers2017computational]. It should be noted that due to
advancements in the CERES algorithm to account for seed effects, the
RNAi dependency has been rendered redundant, and further data releases
for this dependency measurement have been discontinued as of `19Q3`
[@broad2020depmap; @dempster2019extracting]. In addition genomic
dependency measurements of cancer cell lines, chemical dependencies
were taken via Depmap PRISM viability screens that as of the `20Q1`
release, tested 4,518 compounds against 578 cancer cell lines
[@corsello2019non; @broad2020depmap]. The Depmap project has also
compiled additional datasets detailing molecular biological
characterization of cancer cell lines, such as genomic copy number,
Reverse Phase Protein Array (RPPA) data, TPM gene expression data for
protein coding genes and genomic mutation data. These datasets are
updated quarterly on a release schedule and are publically available
under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)
licence [@broad2020depmap].

Table of all depmap datasets

```{r echo=FALSE, results='asis'}
library("knitr")
kable(stuff, caption = "Table of all depmap datasets")
```


The `depmap` Bioconductor package was created in order to maximally
exploit these rich datasets and to aide reproducible research, by
importing the data into the R environment. The Depmap datasets were
cleaned by converting all datasets to the long format, as well as
adding the unique key `depmap_id` for all data tables, in order to
make features more comparable, facilitating the use of common R
packages such as `dplyr` [@wickham2020package] and `ggplot2`
[@wickham2011ggplot2].

As new Depmap datasets are released on a quarterly basis, it is not
feasible to include all dataset files in binary directly within the
directory of the `depmap` R package. To keep the package lightweight,
the `depmap` package utilizes and fully depends on the `ExperimentHub`
package [@ExperimentHub] to store and retrieve all versions of the
Depmap data (starting from `19Q1` through `20Q1`) in the Cloud using
AWS. The `depmap` package contains accessor functions to directly
download and cache the most current datasets from the Cloud into the
local R environment. Specific datasets, such as older datasets, which
have been used in prior research can also be downloaded, if
desired. This feature has the added advantage of enhancing
reproducible research, such that specific versions of Depmap data can
be selected, in addition to having access to the most current
datasets. The `depmap` R package is available as part of Bioconductor
at:
[https://bioconductor.org/packages/depmap](https://bioconductor.org/packages/depmap).

# Use cases

The features of primary interest from the Depmap Project are the measurements of
cancer dependency scores, found in datasets `crispr` and `rnai`, which
illustrate genetic dependency and the dataset `drug_sensitivity`, which
illustrates chemical dependency. In the case of genetic dependency, the
dependency score is an expression of how vital a particular gene for a given
cancer cell line is in terms of the lethality resulting from the knockout or
knockdown of that gene. For example, a highly negative dependency score is
derived from a large negative log fold change in the population of cancer cells
after gene knockout or knockdown, implying that a given cell line is highly
dependent on that gene. Genes that possess such highly negative dependency
scores may be interesting targets for research in cancer medicine.

Below, we start by loading the packages need to run this workflow.

```{r load_libraries, message=FALSE, warning=FALSE, echo=TRUE}
library("depmap")
library("ExperimentHub")
library("dplyr")
library("ggplot2")
library("stringr")
```

The `depmap` datasets are too large to be included into a typical
package, therefore these data are stored in the Cloud. There are two
ways to access the `depmap` datasets. The first such way calls on
dedicated accessor functions that download, cache and load the latest
available dataset into the R workspace. Examples for all available
data are shown below:

```{r EH1, eval=FALSE}
rnai <- depmap_rnai()
crispr <- depmap_crispr()
copyNumber <- depmap_copyNumber()
TPM <- depmap_RPPA()
RPPA <- depmap_TPM()
metadata <- depmap_metadata()
mutationCalls <- depmap_mutationCalls()
drug_sensitivity <- depmap_drug_sensitivity()
```

Alternatively, specific dataset (from any available release) can be
accessed through Bioconductor's `ExperimentHub`. The `ExperimentHub()`
function creates an `ExperimentHub` object, which can be queried for
specific terms. The list of datasets available that correspond to the
query, `depmap` are shown below:

```{r EH2, message=FALSE, warning=FALSE, echo=TRUE}
## create ExperimentHub query object
eh <- ExperimentHub()
query(eh, "depmap")
```

Specific datasets are downloaded, cached and loaded into the workspace
by selecting each dataset by their unique `EH` numbers. Shown below, datasets
from the `19_Q3` release are downloaded in this way.

```{r load_data, message=FALSE, warning=FALSE, echo=TRUE}
## download and cache required datasets 
metadata <- eh[["EH3086"]]
crispr <- eh[["EH3081"]]
TPM <- eh[["EH3084"]]
mutationCalls <- eh[["EH3085"]]
copyNumber <- eh[["EH3082"]]
```

By importing the `depmap` data into the R environment, the data can be
mined more effectively. For example, if one was interested in
researching soft tissue sarcomas and wanted to search all such cancer
cell lines for the gene with the greatest dependency, one could
accomplish this task by using functions from the `dplyr`
package. Below, the `crispr` dataset is selected for cell lines with
*"SOFT_TISSUE"* in the CCLE name, and displaying a list of the highest
dependency scores.

```{r, soft_tissue_cell_lines, echo=TRUE}
## list of dependency scores
crispr %>% 
  dplyr::select(cell_line, gene_name, dependency) %>%
  dplyr::filter(stringr::str_detect(cell_line, "SOFT_TISSUE")) %>%
  dplyr::arrange(dependency)
```

The gene `RPL14` appears several times in the top dependencies scores,
and may make an interesting candidate target. Figure
\@ref(fig:histrpl14) displays the `crispr` data as a histogram showing
the distribution of dependency scores for gene `RPL14`. The red dotted
line signifies the mean dependency score for that gene, while the blue
dotted line signifies the global mean dependency score for all
`crispr` measurements.

```{r histrpl14, message=FALSE, warning=FALSE, fig.cap="Histogram of CRISPR dependency scores for gene RPL14."}
mean_crispr_dep <- crispr %>% 
  dplyr::select(gene_name, dependency) %>% 
  dplyr::filter(gene_name == "RPL14")

crispr %>% 
  dplyr::select(gene, gene_name, dependency) %>% 
  dplyr::filter(gene_name == "RPL14") %>% 
  ggplot(aes(x = dependency)) + geom_histogram() +
  geom_vline(xintercept = mean(mean_crispr_dep$dependency, na.rm = TRUE), 
             linetype = "dotted", color = "red") +
  geom_vline(xintercept = mean(crispr$dependency, na.rm = TRUE),
             linetype = "dotted", color = "blue")
```

A more complex plot of the `crispr` data, as shown below involves plotting the
distribution of dependency scores for gene `RPL14` for each major type of
cancer, while highlighting the nature of mutations of this gene in such cancer
cell lines (e.g. if such mutations are damaging, etc.). Notice that the
plot above reflects the same overall distribution in two dimensions.

```{r rpl14, message=FALSE, warning=FALSE, fig.cap="Plot of CRISPR dependency scores for gene RPL14 by lineage."}
meta_crispr <- metadata %>%
  dplyr::select(depmap_id, lineage) %>%
  dplyr::full_join(crispr, by = "depmap_id") %>%
  dplyr::filter(gene_name == "RPL14") %>% 
  dplyr::full_join((mutationCalls %>%
                      dplyr::select(depmap_id, entrez_id, 
                                    is_cosmic_hotspot, 
                                    var_annotation)), 
                   by = c("depmap_id", "entrez_id"))

meta_crispr %>%
    ggplot(aes(x = dependency, y = lineage)) +
    geom_point(alpha = 0.4, size = 0.5) +
    geom_point(data = subset(meta_crispr,
                             var_annotation == "damaging"), 
               color = "red") +
    geom_point(data = subset(meta_crispr,
                             var_annotation == "other non-conserving"),
               color = "blue") +
    geom_vline(xintercept = mean(meta_crispr$dependency, na.rm = TRUE),
               linetype = "dotted", color = "red") +
    geom_vline(xintercept = mean(crispr$dependency, na.rm = TRUE),
               linetype = "dotted", color = "blue") 
```

Many cancer phenotypes are the result of changes in gene expression
[@li2019changes; @hernandez2019many; @felts2019stochastic]. The
extensive coverage of the `depmap` data affords visualization of
genetic expression patterns across many major types of cancer. Figure
\@ref(fig:tpm) below shows a boxplot illustrating expression values
for gene `RPL14` by lineage:

```{r tpm, message=FALSE, warning=FALSE, fig.cap="Boxplot of TPM expression values for gene RPL14 by lineage."}
metadata %>%
  dplyr::select(depmap_id, lineage) %>%
  dplyr::full_join(TPM, by = "depmap_id") %>%
  dplyr::filter(gene_name == "RPL14") %>% 
  ggplot(aes(x = lineage, y = expression, fill = lineage)) +
  geom_boxplot(outlier.alpha = 0.1) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none")
```

Differentially expressed genes and elevated genetic dependency in
cancer cell lines have been observed [@tsherniak2017defining;
@mcfarland2018improved]. Therefore, such genes may present themselves
as interesting research targets. Figure \@ref(fig:rhabdomyosarcoma2)
shows a plot of expression versus CRISPR gene dependency for
Rhabdomyosarcoma. The red vertical line represents the average
expression for this form of cancer, while the horizontal line
represents the average dependency for this cancer.

```{r rhabdomyosarcoma1, message=FALSE, warning=FALSE}
sarcoma <- metadata %>%
  dplyr::select(depmap_id, cell_line, primary_disease, subtype_disease) %>%
  dplyr::filter(primary_disease == "Sarcoma",
                subtype_disease == "Rhabdomyosarcoma")

crispr_sub <- crispr %>% 
  dplyr::select(depmap_id, gene, gene_name, dependency)

tpm_sub <- TPM %>% 
  dplyr::select(depmap_id, gene, gene_name, expression)

sarcoma_dep <- sarcoma %>% 
  dplyr::left_join(crispr_sub, by = "depmap_id") %>%
  dplyr::select(-cell_line, -primary_disease,
                -subtype_disease, -gene_name)

sarcoma_exp <- sarcoma %>% 
  dplyr::left_join(tpm_sub, by = "depmap_id")

sarcoma_dat_exp <- dplyr::full_join(sarcoma_dep, sarcoma_exp,
                                    by = c("depmap_id", "gene")) %>%
                   dplyr::filter(!is.na(expression))
```

```{r rhabdomyosarcoma2, message=FALSE, warning=FALSE, fig.cap = "Expression vs crispr gene dependency for Rhabdomyosarcoma."}
sarcoma_dat_exp %>% 
ggplot(aes(x = dependency, y = expression)) + 
  geom_point(alpha = 0.4, size = 0.5) +
  geom_vline(xintercept = mean(sarcoma_dat_exp$dependency, na.rm = TRUE),
             linetype = "dotted", color = "red") +
  geom_hline(yintercept = mean(sarcoma_dat_exp$expression, na.rm = TRUE),
             linetype = "dotted", color = "red") +
  ggtitle("Scatterplot of CRISPR dependency vs expression values for gene") +
  theme(axis.text.x = element_text(angle = 45))
```

Genes with the lowest depenency scores and highest TPM gene expression
are found in the upper left section of the plot above
\@ref(fig:rhabdomyosarcoma2). Such genes, shown below, may present an example
as interesting research targets.

```{r}
sarcoma_dat_exp %>%
  dplyr::select(cell_line, gene_name, dependency, expression) %>%
  dplyr::arrange(dependency, expression)
```

Changes in genomic copy number also play a role in some cancer
phenotypes [@meyers2017computational; @aguirre2016genomic;
@shao2019copy]. The `depmap` data allows the display of log genomic
copy number for across many cancer lineages. Figure
\@ref(fig:boxrpl14) shows such a plot for gene `RPL14` for each major type
of cancer lineage:

```{r boxrpl14, message=FALSE, warning=FALSE, fig.cap="Boxplot of log copy number for gene RPL14 by lineage"}
metadata %>%
  dplyr::select(depmap_id, lineage) %>%
  dplyr::full_join(copyNumber, by = "depmap_id") %>%
  dplyr::filter(gene_name == "RPL14") %>%
  ggplot(aes(x = lineage, y = log_copy_number, fill = lineage)) +
  geom_boxplot(outlier.alpha = 0.1) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  theme(legend.position = "none")
```

# Discussion and outlook

We hope that this package will be used by cancer researchers to dig deeper into
the Depmap data and to support their research. Additionally, we highly encourage
future `depmap` users to combine `depmap` data with other datasets of interest,
such as `TCGA` and `CCLE`.

The depmap R package will continue to be maintained in line with the biannual
Bioconductor release, in addition to quarterly releases of Depmap data.

We welcome feedback and questions from the community. We also highly appreciate
contributions to the code in the form of pull requests.

# Software availability

All packages used in this workflow are available from the
Comprehensive R Archive Network (https://cran.r-project.org) or
Bioconductor (http://bioconductor.org). The specific version numbers
of R and the packages used are shown below.

To install the `depmap` package:

```{r message=FALSE, eval=FALSE}
BiocManager::install('depmap')
```

```{r echo = FALSE}
sessionInfo()
```

# Acknowledgements

# Competing interests

No competing interests were disclosed.

# Grant information

# References

